client/src/components/Format/Format.component.tsx

```diff
@@ -0,0 +1,109 @@
+}
+
+const NUMBER_FORMAT_MAP = {
+  // 123,456,789.98
+  amount: (v: FormatValue, locale: string) => Intl.NumberFormat(locale, { maximumFractionDigits: 2 }).format(Number(v)),
+  // 123,456,789.98 %
+  percent: (v: FormatValue, locale: string) =>
+    Intl.NumberFormat(locale, { style: 'percent', maximumFractionDigits: 2 }).format(Number(v)),
+  // 123.46 M$  - K, M, B
+  currency: (v: FormatValue, locale: string, currency: string) =>
+    Intl.NumberFormat(locale, { style: 'currency', currency: currency, notation: 'compact', maximumFractionDigits: 2 }).format(Number(v)),
+  // 123.46 TB  - B, KB, MB, GB, TB
+  size: (v: FormatValue, locale: string) => {
+    if (Number(v) < 1024) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'byte', maximumFractionDigits: 2 }).format(Number(v))
+    }
+    if (Number(v) >= 1024 && Number(v) < 1024**2) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'kilobyte', maximumFractionDigits: 2 }).format(Number(v)/1024)
+    }
+    if (Number(v) >= 1024**2 && Number(v) < 1024**3) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'megabyte', maximumFractionDigits: 2 }).format(Number(v)/1024**2)
+    }
+    if (Number(v) >= 1024**3 && Number(v) < 1024**4) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'gigabyte', maximumFractionDigits: 2 }).format(Number(v)/1024**3)
+    }
+    if (Number(v) >= 1024**4 && Number(v) < 1024**5) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'terabyte', maximumFractionDigits: 2 }).format(Number(v)/1024**4)
+    }
+  },
+  // 123.46 T  - G, K, T,
+  weight: (v: FormatValue, locale: string) => {
+    if (Number(v) < 1000) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'gram', maximumFractionDigits: 2 }).format(Number(v))
+    }
+    if (Number(v) >= 1000 && Number(v) < 1000000) {
+      return Intl.NumberFormat(locale, { style: 'unit', unit: 'kilogram', maximumFractionDigits: 2 }).format(Number(v)/1000)
+    }
+    if (Number(v) >= 1000000) {
+      return Intl.NumberFormat(locale, { maximumFractionDigits: 2 }).format(Number(v)/1000000) + ' t'
+    }
+  },
+}
+
+const DATE_FORMAT_MAP = {
+  day: (v: FormatValue) => Intl.DateTimeFormat('en', {day: '2-digit'}).format(new Date(v)),
+  month: (v: FormatValue) => Intl.DateTimeFormat('en', { month: '2-digit' }).format(new Date(v)),
+  year: (v: FormatValue) => Intl.DateTimeFormat(undefined, { year: 'numeric' }).format(new Date(v)),
+  dayMonthYear: (v: FormatValue) => Intl.DateTimeFormat().format(new Date(v)),
+  dayOfMonthYear: (v: FormatValue) => Intl.DateTimeFormat('en', { day: '2-digit', month: 'long', year: 'numeric' }).format(new Date(v)),
+}
+
+export const FORMAT_MAP = {
+  ...STRING_FORMAT_MAP,
+  ...NUMBER_FORMAT_MAP,
+  ...DATE_FORMAT_MAP,
+}
+
+export type FormatVariant = keyof typeof FORMAT_MAP
+
+export type FormatProps = TextProps & {
```

Comment:
can we add `value?: T | null | undefined` prop and omit `content and children prop from TextProps`?